src/addon/validation.cpp

```diff
@@ -59,12 +87,17 @@ bool addon_name_legal(const std::string& name)
 
 bool addon_filename_legal(const std::string& name)
 {
-	if(name.empty() || name == "." ||
-	   name.find_first_of("/:\\~ \r\n\v\t") != std::string::npos ||
-	   name.find("..") != std::string::npos) {
+	if(name.empty() || name.back() == '.' ||
+	   name.find("..") != std::string::npos ||
+	   name.size() > 255) {
 		return false;
 	} else {
-		return true;
+		const ucs4::string name_ucs4 = unicode_cast<ucs4::string>(name);
+		const std::string name_utf8 = unicode_cast<utf8::string>(name_ucs4);
+		if(name != name_utf8){ // name is invalid UTF-8
```

Comment:
did you test that this check works? I'd think that unicode_cast simply throws an exception in case of invalid utf8

Reply:
See [unicode_cast.hpp](https://github.com/wesnoth/wesnoth/blob/master/src/serialization/unicode_cast.hpp). The cast swallows the exception and returns the part of the string that it was able to convert.

Reply:
I tested this yesterday. It works, but a different check failed in `archive_addon`. Specifically `filesystem::looks_like_pbl` failed because it uses `utf8::lowercase`. I've added a try-catch block for this.